File: pinpointB/src/main/java/com/nxquar/pinpoint/DTO/UpdateUserDto.java
package com.nxquar.pinpoint.DTO;

import com.nxquar.pinpoint.Model.Address;
import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.NoArgsConstructor;

import java.util.UUID;

@Data
@NoArgsConstructor
@AllArgsConstructor
public class UpdateUserDto {
    private String name;
    private String phone;
    private boolean isVerified;
    private Address address;
    private UUID batchId;
    private UUID adminId;
}

File: pinpoint/lib/model/user/update_user_dto.dart
import 'package:json_annotation/json_annotation.dart';
import 'package:pinpoint/model/address/address.dart';
import 'package:pinpoint/model/batch/batch_list_response.dart';

part 'update_user_dto.g.dart';

@JsonSerializable()
class UpdateUserDto {
  final String name;
  final String phone;
  final bool isVerified;
  final Address address;
  final BatchListResponse? batch;
  final String? adminId;

  UpdateUserDto({
    required this.name,
    required this.phone,
    required this.isVerified,
    required this.address,
    this.batch,
    this.adminId,
  });

  factory UpdateUserDto.fromJson(Map<String, dynamic> json) =>
      _$UpdateUserDtoFromJson(json);

  Map<String, dynamic> toJson() => _$UpdateUserDtoToJson(this);
}
